### DISCUSSION TITLE
Sample data

### DISCUSSION THREAD

**Discussion by arjunsavel** [finagle29/DBSP_DRP#28] [Q&A] [answered]
Hi @finagle29 ,

Thanks for pushing the new `develop` branch! With that branch and the new sample data, I'm able to reduce both the blue and red arms. All the QA files look great.

However, I'm still running into a bit difficulty with the `dbsp_ql` command (KeyError: "Keyword 'RA' not found.") when run on the red files. Also, when I run `dbsp_adjust_splicing` on spliced/ZTF20abismub_b.fits, an error (IndexError: index -1 is out of bounds for axis 0 with size 0) is triggered on line 33 of `dbsp_adjust_splicing`; the splicing adjustment works great for the other file (spliced/ZTF20abismub_a.fits). What are your thoughts on these errors?

Thanks in advance!

**Comment by finagle29** [comment, marked as answer]
Thanks for bringing these errors to my attention, these are bugs that I hadn't come across before.

For the ``dbsp_ql`` crash, the red calibration data is missing ra/dec info (which shouldn't be a problem as the telescope dome is closed during calibrations), but PypeIt crashes if all ra/dec data in a reduction are missing. This usually doesn't happen, but when the quicklook first runs and only processes an arc frame and a flat frame, all of the (unimportant) ra/dec data can be missing, leading to the crash. I'm going to file an issue and suggest a fix with PypeIt, but as a workaround for now, the quicklook script will generate fake ra/dec data (zeros) if needed.

What I think is happening with spliced/ZTF20abismub_b.fits is that it only contains data from one side (since the object was not found on the other side) and so no splicing was performed in the initial reduction and therefore the splicing can't be adjusted. I've added a guard statement and error message if ``dbsp_adjust_splicing`` is run on a file that doesn't contain both red and blue side data.

I've addressed these bugs in #29 and will merge that into develop once the required tests pass.
